dotnet/src/Connectors/Connectors.OpenAI.UnitTests/Services/OpenAIChatCompletionServiceTests.cs

```diff
@@ -1106,6 +1106,41 @@ public async Task ItCreatesCorrectFunctionToolCallsWhenUsingNoneFunctionChoiceBe
         Assert.Equal("none", optionsJson.GetProperty("tool_choice").ToString());
     }
 
+    [Fact]
+    public async Task ItCreatesCorrectFunctionToolCallsWhenUsingRequiredFunctionChoiceBehaviorAsync()
+    {
+        // Arrange
+        var kernel = new Kernel();
+        kernel.Plugins.AddFromFunctions("TimePlugin", [
+            KernelFunctionFactory.CreateFromMethod(() => { }, "Date"),
+            KernelFunctionFactory.CreateFromMethod(() => { }, "Now")
+        ]);
+
+        var chatCompletion = new OpenAIChatCompletionService(modelId: "gpt-3.5-turbo", apiKey: "NOKEY", httpClient: this._httpClient);
+
+        using var response = new HttpResponseMessage(HttpStatusCode.OK) { Content = new StringContent(File.ReadAllText("TestData/chat_completion_test_response.json")) };
+        this._messageHandlerStub.ResponseQueue.Enqueue(response);
+
+        var chatHistory = new ChatHistory();
```

Comment:
nit

> [!NOTE]
> I advocate for changing all our lists signature added in this PR to the new format below

```suggestion
        ChatHistory chatHistory = [];
        List<T> t = []; // For other files (invokedFunctions, etc)
```

Reply:
[Out of scope of this PR] Please register an issue to track the ask so it can be triaged and prioritized.